cloudinit/sources/DataSourceLXD.py

```diff
@@ -299,7 +300,17 @@ def _get_json_response(
 def _do_request(
     session: requests.Session, url: str, do_raise: bool = True
 ) -> requests.Response:
-    response = session.get(url)
+    for _ in range(30):
```

Comment:
Is it worth keeping the var around to report how many retries remain?
```suggestion
    for tries in range(30):
    for retries in range(29, 0, -1):
    ....
    LOG.warning(
        "[GET] [HTTP:%d] %s, retrying %d more time(s)", response.status_code, url, retries
```